mne/time_frequency/psd.py

```diff
@@ -140,34 +144,20 @@ def psd_array_welch(x, sfreq, fmin=0, fmax=np.inf, n_fft=256, n_overlap=0,
     # Parallelize across first N-1 dimensions
     x_splits = np.array_split(x, n_jobs)
 
-    if average in ['mean', 'median']:
-        from scipy.signal import welch
-        parallel, my_welch_func, n_jobs = parallel_func(_welch_func,
-                                                        n_jobs=n_jobs)
-
-        psds = parallel(my_welch_func(d, fs=sfreq, freq_mask=freq_mask,
-                                      n_per_seg=n_per_seg, n_overlap=n_overlap,
-                                      nfft=n_fft, average=average, func=welch)
-                        for d in x_splits)
-        psds = np.concatenate(psds, axis=0)
-        psds.shape = dshape + (-1,)
-    elif average is None:
-        from scipy.signal import spectrogram
-        parallel, my_spect_func, n_jobs = parallel_func(_spect_func,
-                                                        n_jobs=n_jobs)
-
-        f_spect = parallel(my_spect_func(d, n_overlap=n_overlap,
-                                         nfft=n_fft,
-                                         fs=sfreq, freq_mask=freq_mask,
-                                         func=spectrogram,
-                                         n_per_seg=n_per_seg)
-                           for d in x_splits)
-        psds = np.concatenate(f_spect, axis=0)
-        psds.shape = dshape + (len(freqs), -1)
-    else:
-        raise ValueError('average must be one of `mean`, `median`, or None, '
-                         'got {}'.format(average))
+    from scipy.signal import spectrogram
+    parallel, my_spect_func, n_jobs = parallel_func(_spect_func, n_jobs=n_jobs)
+
+    f_spect = parallel(my_spect_func(d, n_overlap=n_overlap,
+                                     n_per_seg=n_per_seg, nfft=n_fft, fs=sfreq,
+                                     freq_mask=freq_mask, func=spectrogram)
+                       for d in x_splits)
+    psds = np.concatenate(f_spect, axis=0)
+    psds.shape = dshape + (len(freqs), -1)
 
+    if average == 'mean':
+        psds = np.nanmean(psds, axis=-1)
+    elif average == 'median':
+        psds = np.nanmedian(psds, axis=-1)
```

Comment:
This is going to have a much higher (by a factor of `n_epochs`) memory cost than before. If you write a little `_welch_func` wrapper that does the mean and median (if requested) internally, then loop over that, the cost will go back down.

Reply:
I'm not sure I understand since that's exactly how we did it before we changed to `scipy.signal.welch`. Do you want me to compute a rolling mean/median in a loop?

Reply:
Pushed a commit, hopefully it makes it clearer